archeryutils/targets.py

```diff
@@ -84,53 +100,92 @@ class Target:
 
     """
 
+    _face_spec: Optional[FaceSpec] = None
+
+    supported_systems = get_args(ScoringSystem)
+    supported_distance_units = Length.yard | Length.metre
+    supported_diameter_units = Length.cm | Length.inch | Length.metre
+
     def __init__(
         self,
         scoring_system: ScoringSystem,
         diameter: Union[float, tuple[float, str]],
         distance: Union[float, tuple[float, str]],
         indoor: bool = False,
     ) -> None:
-        systems = get_args(ScoringSystem)
-
-        if scoring_system not in systems:
+        if scoring_system not in self.supported_systems:
             msg = (
                 f"""Invalid Target Face Type specified.\n"""
-                f"""Please select from '{"', '".join(systems)}'."""
+                f"""Please select from '{"', '".join(self.supported_systems)}'."""
             )
-
             raise ValueError(msg)
+        dist, native_dist_unit = Length.parse_optional_units(
+            distance, self.supported_distance_units, "metre"
+        )
+        diam, native_diameter_unit = Length.parse_optional_units(
+            diameter, self.supported_diameter_units, "cm"
+        )
 
-        if isinstance(distance, tuple):
-            (distance, native_dist_unit) = distance
-        else:
-            native_dist_unit = "metre"
-        if native_dist_unit not in Length.yard | Length.metre:
-            msg = (
-                f"Distance unit '{native_dist_unit}' not recognised. "
-                "Select from 'yard' or 'metre'."
-            )
-            raise ValueError(msg)
-        distance = Length.to_metres(distance, native_dist_unit)
+        self._scoring_system = scoring_system
+        self._distance = Length.to_metres(dist, native_dist_unit)
+        self._native_dist_unit = native_dist_unit
```

Comment:
You've lost me slightly on storing as a `Length`, at the moment that is the utility class for conversions and parsing of lengths and units (though I'm looking at making that an instance rather than directly using the class).

But assuming I've got your drift, I do agree it might make more sense to store the `(native_quantity, native_units)` tuples directly and access those as needed rather than building them on the fly from a conversion of the `si_quantity` and `native_units` attributes. Again since these are read_only properties now we don't have to worry about keeping them in sync, and if we do add setters in the future those can contain the logic to keep everything updated.

Also could then propagate those changes up to `Pass`.

I still quite like the idea of a NamedTuple, heres an example from one the tests where I introduced it which shows how it makes things a bit more readable:
```python
imperial_target = Target(
    "Worcester",
    (16, "Inches"),
    (20, "Yards"),
    indoor=True,
)

assert imperial_target.distance == 18.288

# before this pr:
# can get the unit two different ways
assert imperial_target.native_distance_unit == "yard"
assert imperial_target.native_distance == (20, "yard")

# if we remove native_distance_unit attribute options to get the unit are:
# simple tuple access
assert imperial_target.native_distance[0] == "yard"

# unpacking
_, unit = imperial_target.native_distance
assert unit == "yard"

# named tuple access looks cleaner and still supports unpacking
assert imperial_target.native_distance.units == "yard"
```

Reply:
In fact (helping to make my case), I used the wrong index on the tuple there, it should have been
```python
assert imperial_target.native_distance[1] == "yard"
```

Reply:
Yeah, forget that, sorry - I forgot the `Length` class doesn't store magnitude and unit, just acts as a converter.
What I was envisaging is basically what you have here with `Q` I guess.

Though that makes me wonder if `Length` should hold a magnitude and be what is passed around internally?
Or perhaps that would get in the way of users being able to supply just a numerical value...

Reply:
I think we're already close to best compromise here: return Quantities for the nice experience, but accept plain numeric values or regular tuples. 
Have now implemented storing this directly and using it to looking up from when the units are wanted.